packages/block-editor/src/hooks/background.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * WordPress dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { getBlockSupport } from '@wordpress/blocks';
+import {
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	Button,
+	DropZone,
+	FlexItem,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalHStack as HStack,
+	__experimentalTruncate as Truncate,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { Platform, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { getFilename } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../components/inspector-controls';
+import MediaReplaceFlow from '../components/media-replace-flow';
+import MediaUpload from '../components/media-upload';
+import MediaUploadCheck from '../components/media-upload/check';
+import useSetting from '../components/use-setting';
+import { cleanEmptyObject } from './utils';
+import { store as blockEditorStore } from '../store';
+
+export const BACKGROUND_SUPPORT_KEY = 'background';
+export const IMAGE_BACKGROUND_TYPE = 'image';
+
+/**
+ * Checks if there is a current value in the background image block support
+ * attributes.
+ *
+ * @param {Object} props Block props.
+ * @return {boolean}     Whether or not the block has a background image value set.
+ */
+export function hasBackgroundImageValue( props ) {
+	const hasValue =
+		!! props.attributes.style?.background?.backgroundImage?.id ||
+		!! props.attributes.style?.background?.backgroundImage?.url;
+
+	return hasValue;
+}
+
+/**
+ * Determine whether there is block support for background.
+ *
+ * @param {string} blockName Block name.
+ * @param {string} feature   Background image feature to check for.
+ *
+ * @return {boolean} Whether there is support.
+ */
+export function hasBackgroundSupport( blockName, feature = 'any' ) {
+	if ( Platform.OS !== 'web' ) {
+		return false;
+	}
+
+	const support = getBlockSupport( blockName, BACKGROUND_SUPPORT_KEY );
+
+	if ( support === true ) {
+		return true;
+	}
+
+	return !! support?.[ feature ];
```

Comment:
Good catch, yes, I'd removed the check because we only have one feature for now, but we should still handle the `any` case. I've added a check back in, which gives us a natural place to put other support features in follow-ups when we add in additional controls 👍